doc/radosgw/cloud-sync-module.rst

```diff
@@ -0,0 +1,254 @@
+and data is not synced from the remote zone back. The aim of this sync module is to
+provide capability of syncing data to different cloud providers. Currently the supported
+cloud providers are ones that are compatible with AWS (S3).
+
+A user for the remote cloud object store service needs to be configured. Sync operations will
+be done under that speicified user. Since different cloud services impose limits on the number
+of buckets that each user can create, the source objects and buckets will be mapped into a
+different (configurable) buckets and objects. It is possible to configure different targets
+to different buckets and bucket prefixes. In addition to that, source ACLs will not be preserved.
+It is possible to map permission to specific source users to a specific destination users.
+
+Due to API limitations, there is no way to preserve original objects modification time, and
+ETag. The cloud sync module stores these in a separate metadata attributes on the destination
+objects.
+
+
+
+Cloud Sync Tier Type Configuration
+-------------------------------------
+
+Trivial Configuration:
+~~~~~~~~~~~~~~~~~~~~~~
+
+::
+    {
+      "connection": {
+        "access_key": <access>,
+        "secret": <secret>,
+        "endpoint": <endpoint>,
+        "host_style": <path | virtual>,
+      },
+      "acls": [ { "type": <id | email | uri>,
+                  "source_id": <source_id>,
+                  "dest_id": <dest_id> } ... ],
+      "target_path": <target_path>,
+    }
+
+
+Non Trivial Configuration:
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+    {
+      "default": {
+        "connection": {
+            "access_key": <access>,
+            "secret": <secret>,
+            "endpoint": <endpoint>,
+            "host_style" <path | virtual>,
+        },
+        "acls": [
+        {
+          "type" : <id | email | uri>,   #  optional, default is id
+          "source_id": <id>,
+          "dest_id": <id>
+        } ... ]
+        "target_path": <path> # optional
+      },
+      "connections": [
+          {
+            "connection_id": <id>,
+            "access_key": <access>,
+            "secret": <secret>,
+            "endpoint": <endpoint>,
+            "host_style" <path | virtual>,  # optional
+          } ... ],
+      "acl_profiles": [
+          {
+            "acls_id": <id>, # acl mappings
+            "acls": [ {
+                "type": <id | email | uri>,
+                "source_id": <id>,
+                "dest_id": <id>
+              } ... ]
+          }
+      ],
+      "profiles": [
+          {
+           "source_bucket": <source>,
+           "connection_id": <connection_id>,
+           "acls_id": <mappings_id>,
+           "target_path": <dest>,          # optional
+          } ... ],
+    }
+
+
+.. Note:: Trivial configuration can coincide with the non-trivial one.
+
+
+* ``endpoint``
+
+Specifies the Elasticsearch server endpoint to access
+
+* ``connection`` (container)
```

Comment:
looks like this was copy/pasted from Elasticsearch. but the 'endpoint' field is properly documented below under 'connection', so this one can be removed?

Reply:
@cbodley I'll fix it, thanks. Shouldn't hold merge though